pkg/sql/sem/tree/datum.go

```diff
@@ -985,6 +1016,11 @@ func (*DBytes) ResolvedType() types.T {
 	return types.Bytes
 }
 
+// AsJSON implements transformation to JSON
+func (d *DBytes) AsJSON() (json.JSON, error) {
```

Comment:
This one will require some tests to ensure the escaping for various characters (like non-printing and valid/invalid unicode characters) matches Postgres' behaviour.